src/index.js

```diff
@@ -0,0 +1,232 @@
+
+let equation = {
+    flag : 0,
+    pows : {
+    }
+};
+
+main();
+
+function main() {
+    let a = 0;
+    let b = 0;
+    let c = 0;
+    let discrim = 0;
+    let x = [];
+    if (process.argv.length != 3 || process.argv[2].length == 0) {
+        return errors(0);
+    }
+    // console.log(process.argv[2]);
+    const equat = process.argv[2];
+    equatValid(equat);
+
+    if (checkParams(equat) < 0 || reducedForm(equat) < 0)
+        return (-1);
+    a = (equation.pows[2]) ? equation.pows[2] : 0;
+    b = (equation.pows[1]) ? equation.pows[1] : 0;
+    c = (equation.pows[0]) ? equation.pows[0] : 0;
+    if (a === 0 ) {
+        x.push(-c / b);
+    } else {
+        discrim = b * b - 4 * a * c;
+        x = getRoots(a, b, c, discrim);
+    }
+    if (discrim === 0 || (a === 0 && discrim >= 0)) {
+        console.log("The solution is: ")
+    } else if (discrim < 0) {
+        console.log("Discriminant is negative, but I can solve it: ");
+    } else {
+        console.log("Discriminant is strictly positive, the two solutions are: ");
+    }
+    for (res in x) {
+        console.log(x[res]);
+    }
+}
+
+function equatValid(equat) {
+    // console.log('equat ', equat);
+    let re = /[0-9X*+^=. -]+$/;
+    let res = equat.search(re);
+
+    if (res === -1 || !equat.includes('=')) {
+        console.log("look here");
+        errors(1);
+    }
+    return 1;
+}
+
+function numberValid(equat) {
+    let re = /^[\d.]+$/;
+    let res = equat.search(re);
+
+    console.log("func number valid");
+    if (res === -1) {
+        console.log("func number not valid");
+        return (errors(1));
+    }
+    return equat;
```

Comment:
you return what u get?
numberValid function name says that from here u get true or false